src/librustc_typeck/check/mod.rs

```diff
@@ -1709,6 +1722,10 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
     }
 
     pub fn expr_ty(&self, ex: &hir::Expr) -> Ty<'tcx> {
+        if let Some(&adjustment::AdjustEmptyToAny(ref t))
+                = self.tables.borrow().adjustments.get(&ex.id) {
+            return t;
+        }
```

Comment:
Huh. Kind of wacky that we only consider *this* kind of adjustment -- do we believe that this will be the only case in which an adjustment is present?